olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -681,5 +747,82 @@ std::string DefaultCacheImpl::GetExpiryKey(const std::string& key) const {
   return CreateExpiryKey(key);
 }
 
+bool DefaultCacheImpl::Protect(const DefaultCache::KeyListType& keys) {
+  std::lock_guard<std::mutex> lock(cache_lock_);
+  auto start = std::chrono::steady_clock::now();
+  protected_keys_.Protect(keys, [&](const std::string& key) {
+    if (!RemoveKeyLru(key)) {
+      RemoveKeysWithPrefixLru(key);
+    }
+  });
+  if (memory_cache_) {
+    memory_cache_->Clear();
+  }
+
+  OLP_SDK_LOG_INFO_F(kLogTag, "Protect, time=%" PRId64 " ms",
+                     GetElapsedTimeFromStart(start));
+  return true;
+}
+
+bool DefaultCacheImpl::Release(const DefaultCache::KeyListType& keys) {
+  std::lock_guard<std::mutex> lock(cache_lock_);
+  auto start = std::chrono::steady_clock::now();
+  bool need_reset = false;
+  auto result = protected_keys_.Release(keys, [&](const std::string& key) {
+    if (!need_reset) {
+      KeyValueCache::ValueTypePtr value = nullptr;
+      time_t expiry = KeyValueCache::kDefaultExpiry;
+
+      auto result = GetFromDiskCache(key, value, expiry);
+      if (result && value) {
+        if (memory_cache_) {
+          if (expiry > 0) {
+            memory_cache_->Put(key, value, expiry, value->size());
+          } else {
+            memory_cache_->Remove(key);
+          }
+        }
+        if (expiry > 0) {
```

Comment:
Why does the expiry matter to put it in the LRU cache? If a key was released from protection we should add it to LRU either way so that it can be removed when DB size is growing. Is there a reason to only put it if it still has expiry left?